modules/storage-api/src/main/java/org/apache/ignite/internal/storage/MvPartitionStorage.java

```diff
@@ -151,7 +151,7 @@ public interface MvPartitionStorage extends ManuallyCloseable {
      * @throws TxIdMismatchException If there's another pending update associated with different transaction id.
      * @throws StorageException If failed to write data to the storage.
      */
-    @Nullable BinaryRow addWrite(RowId rowId, @Nullable BinaryRow row, UUID txId, UUID commitTableId, int commitPartitionId)
+    @Nullable TableRow addWrite(RowId rowId, @Nullable TableRow row, UUID txId, UUID commitTableId, int commitPartitionId)
```

Comment:
Why not use **BinaryTuple**? This change looks strange.

Reply:
table can change over time, so we must store the version of the schema that was used to build the tuple so we can read its contents later

Reply:
Within a transaction at any given moment of time only one table schema is in effect. It does not change arbitrarily. It is redundant to specify schema for every inserted row. The storage engine does not have to store the schema with every row. It is sufficient to remember the transaction timestamps since which a given table schema version is in effect.

Reply:
> The storage engine does not have to store the schema version together with every row too. It is sufficient to remember the transaction timestamp since which a given table schema version is in effect.

That is not how it's working right now.

Reply:
Columnar storage does not store schema version with rows right now.

The whole effort was triggered because BinaryRow had too many assumptions that were plain wrong for columnar. Now this PR adds TableRow that basically has the same problem.

Schema version should be handled in the same fashion as rowid is handled now. That is it should be passed as an additional parameter or field or whatever. It should never be embedded into the binary data. This way we will avoid troubles for storage engines that do not need it.

Reply:
Ignite does not have columnar storage